### PR TITLE
Adding support for SolarLog EMS

### PR DESCRIPTION
**Pull request by dschuesae**

### CODE REVIEW

File: docs/modules/EMS_SolarLog.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+Please note, if any of the required parameters for the SolarLog EMS module are not specified in the module configuration, the module will unload at start time.
+
+## JSON Configuration Example
+
+```
+"SolarLog": {
+  "enabled": false,
+  "serverIP": "192.168.1.2",
+  "excludeConsumptionInverters": "2"
```
Comment (MikeBishop) [inline docs/modules/EMS_SolarLog.md:25]: From your code, it looks like you expect this to be a comma-separated list.  An array might be clearer, and if they're integers you can skip the quotes.

Either way, I'd have multiple values in the example if you support it; it's easy to simplify a multi-value example to a single value, but generalizing a single example to multiple values requires reading code.

File: lib/TWCManager/EMS/SolarLog.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    excludeConsumptionInverters = {}
+    timeout = 2
+
+    def __init__(self, master):
+        self.master = master
+        self.config = master.config
+        try:
+            self.configConfig = master.config["config"]
+        except KeyError:
+            self.configConfig = {}
```
Comment (MikeBishop) [inline lib/TWCManager/EMS/SolarLog.py:31]:
```suggestion
        self.configConfig = master.config.get("config", {})
```

File: lib/TWCManager/EMS/SolarLog.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        try:
+            self.configSolarLog = master.config["sources"]["SolarLog"]
+        except KeyError:
+            self.configSolarLog = {}
```
Comment (MikeBishop) [inline lib/TWCManager/EMS/SolarLog.py:35]:
```suggestion
        self.configSolarLog = master.config["sources"].get("SolarLog", {})
```

File: lib/TWCManager/EMS/SolarLog.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        self.cacheTime = cacheTime
+
+    def setTimeout(self, timeout):
+        self.timeout = timeout
+
+    def update(self):
+        # Update function - determine if an update is required
+
+        if (int(self.time.time()) - self.lastFetch) > self.cacheTime:
+            # Cache has expired. Fetch values from HomeAssistant sensor.
```
Comment (MikeBishop) [inline lib/TWCManager/EMS/SolarLog.py:171]:
```suggestion
            # Cache has expired. Fetch values from SolarLog.
```

File: docs/modules/EMS_SolarLog.md (hunk 1 above)
Comment (dschuesae) [reply docs/modules/EMS_SolarLog.md]: Completely agree. Just updated the branch.

**Pull request merged**